src/components/shared/list-item-swipe-action/ListItemSwipeAction.tsx

```diff
@@ -12,138 +12,135 @@ interface IListItemSwipeActionProps {
+      const swipableWidth = frontDivRef.current.getBoundingClientRect().width;
+      const triggerDelete = currentPosition / swipableWidth > 0.25;
+      const positionTarget = triggerDelete ? swipableWidth : 0;
+
+      const animateObject = { position: currentPosition };
+      frontDivAnimationRef.current = anime({
+        complete: () => {
+          if (triggerDelete) {
+            onAction();
+          }
+        },
+        duration: 250,
+        easing: "linear",
+        position: positionTarget,
+        targets: animateObject,
+        update: () => {
+          setLastPosition(animateObject.position);
+          setTranslateX(`${animateObject.position}px`);
+        },
+      });
+    }
 
-  private onPan(event: HammerInput) {
-    const { translateX, lastPosition } = this.state;
-    const { deltaX } = event;
+    function onPan(event: HammerInput) {
+      const { deltaX } = event;
 
-    if (this.frontDivAnimation) {
-      this.frontDivAnimation.pause();
-      this.frontDivAnimation = null;
-    }
+      if (frontDivAnimationRef.current) {
+        frontDivAnimationRef.current.pause();
+        frontDivAnimationRef.current = null;
+      }
 
-    let frontPosition = lastPosition + deltaX;
+      let frontPosition = lastPositionRef.current + deltaX;
 
-    if (frontPosition < 0) {
-      frontPosition = 0;
-    }
+      if (frontPosition < 0) {
+        frontPosition = 0;
+      }
 
-    if (event.isFinal) {
-      this.onFinal(frontPosition);
-    } else {
-      this.setState({
-        translateX: `${frontPosition}px`,
-      });
+      if (event.isFinal) {
+        onFinal(frontPosition);
+      } else {
+        setTranslateX(`${frontPosition}px`);
+      }
     }
-  }
-
-  private onFinal(currentPosition: number) {
-    const swipableWidth = this.frontDiv!.getBoundingClientRect().width;
-    const triggerDelete = currentPosition / swipableWidth > 0.25;
-    const positionTarget = triggerDelete ? swipableWidth : 0;
-
-    const animateObject = { position: currentPosition };
-    this.frontDivAnimation = anime({
-      complete: () => {
-        if (triggerDelete) {
-          this.onAction();
-        }
-      },
-      duration: 250,
-      easing: "linear",
-      position: positionTarget,
-      targets: animateObject,
-      update: () => {
-        this.setState({
-          lastPosition: animateObject.position,
-          translateX: `${animateObject.position}px`,
-        });
-      },
-    });
-  }
-
-  private onAction() {
-    const animateObject = {
-      height: this.frontDiv!.clientHeight,
-      opacity: 1,
+    mcFrontDivRef.current = new Hammer(frontDivRef.current);
+
+    mcFrontDivRef.current
+      .get("pan")
+      .set({ direction: Hammer.DIRECTION_HORIZONTAL });
+    mcFrontDivRef.current.on("pan", onPan);
+
+    return () => {
+      mcFrontDivRef.current?.destroy();
     };
+  }, [lastPositionRef, onActionRef]);
+
+  return { frontDivRef, height, opacity, translateX } as const;
```

Comment:
This whole hook is very similar to [what I did for `SwipeableModal`](https://github.com/OpenSourceRaidGuild/atom-pwa/blob/next/src/components/shared/swipeable-modal/SwipeableModal.tsx#L12-L117).  I wonder if we can pull out a common hook and use it in both?

Reply:
I was thinking th same thing for the questions logic on the tests, we can see if thats possible in following prs? :)

Reply:
agree it might be nice, but out of scope for  this PR which is already 🤯